psa-crypto-sys/src/shim_methods.rs

```diff
@@ -171,3 +171,15 @@ pub fn PSA_KEY_TYPE_DH_KEY_PAIR(group: psa_dh_group_t) -> psa_key_type_t {
 pub fn PSA_KEY_TYPE_DH_PUBLIC_KEY(group: psa_dh_group_t) -> psa_key_type_t {
     unsafe { psa_crypto_binding::shim_PSA_KEY_TYPE_DH_PUBLIC_KEY(group) }
 }
+
+pub fn PSA_SIGN_OUTPUT_SIZE(
```

Comment:
I am thinking if those macros should be safe or not... The specification says:

> If the parameters are not valid, the return value is unspecified.

and I am wondering if `unspecified` is also `unsafe` 😋 Maybe it is safer to add `unsafe` here, ironically

Reply:
Can't we do enough checks to make it safe? As in, if we have the checks in place to ensure we only ever pass the macros valid arguments, it shouldn't ever return an unspecified value.

Reply:
> Can't we do enough checks to make it safe?

I definitely agree and that is what we are trying to do in the `psa-crypto` crate. The `psa-crypto-sys` should only be the thin Rust wrapper to allow calling the C API and can be unsafe.

Reply:
Ah ok, I see what you mean now. Yes I think they should be marked as unsafe as you suggested in `psa-crypto-sys`.